src/ledger/LedgerTxn.cpp

```diff
@@ -372,6 +400,17 @@ LedgerTxn::Impl::deactivate(LedgerKey const& key)
     {
         throw std::runtime_error("Key is not active");
     }
+
+    auto entryIter = mEntry.find(key);
```

Comment:
could we maybe rewrite those code changes into a call to a method `updateOrderBook` or sorts that can by itself reason about `mActive` and `mEntry` vs `mOrderBook`. That way it's easier to understand which invariant are supposed to be met. Right now it's pretty hard to see if we've missed an update.
Basically, if we could just call a method every time we update `mActive` or `mEntry` that ensures that the in memory order book has the right state the change would be much easier to validate.